spec/features/results_spec.rb

```diff
@@ -226,19 +231,58 @@
         {
           l: 3,
           query: "",
-          qualifications: "QtsOnly,PgdePgceWithQts,Other",
-          fulltime: "False",
-          parttime: "False",
-          hasvacancies: "True",
-          senCourses: "False",
+          qualifications: %w[QtsOnly PgdePgceWithQts Other],
+          fulltime: "false",
+          parttime: "false",
+          hasvacancies: "true",
+          senCourses: "false",
         }
       end
 
       it "falls back to l2 (Across England)" do
         location_filter_uri = URI(current_url)
         expect(location_filter_uri.path).to eq("/results")
-        expect(location_filter_uri.query).to eq("fulltime=False&hasvacancies=True&l=2&parttime=False&qualifications=QtsOnly%2CPgdePgceWithQts%2COther&senCourses=False")
+        expect(Rack::Utils.parse_nested_query(location_filter_uri.query)).to eq({
+          "fulltime" => "false",
+          "hasvacancies" => "true",
+          "l" => "2",
+          "parttime" => "false",
+          "qualifications" => %w[QtsOnly PgdePgceWithQts Other],
+          "senCourses" => "false",
+        })
       end
     end
   end
+
+  context "with C# parameters" do
+    let(:base_parameters) { results_page_parameters("sort" => sort, "filter[subjects]" => "C1,08,F1") }
+
+    let(:params) do
+      {
+        l: 2,
+        query: "",
+        qualifications: "QtsOnly,PgdePgceWithQts,Other",
+        subjects: "1,2,3",
+        fulltime: "False",
+        parttime: "False",
+        hasvacancies: "True",
+        senCourses: "False",
+      }
+    end
+
+    it "redirects to rails parameters" do
```

Comment:
It would be good if we could get rid of all these query checking tests out of the feature specs now the results page is functional. These should be features from a users perspective so we should really test the state of the UI or the results rather than querystrings. Is there another way we could test this? (I realise there are loads of them but we need to start somewhere :) )